packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -15,6 +15,33 @@ import 'navigation_bar_theme.dart';
 import 'theme.dart';
 import 'tooltip.dart';
 
+const double _kBadgeSize = 5.0;
+
+const double _kBadgeLabelHeight = 30.0;
+const double _kBadgeLabelWidth = 35.0;
+
+const Offset _kBadgeTopStart = Offset(30.0, 6.0);
+const Offset _kBadgeTopEnd = Offset(50.0, 6.0);
+const Offset _kBadgeBottomStart = Offset(30.0, 26.0);
+const Offset _kBadgeBottomEnd = Offset(50.0, 26.0);
+
+const Offset _kBadgeLabelTopStart = Offset(20.25, 7.0);
+const Offset _kBadgeLabelTopEnd = Offset(60.0, 7.0);
+const Offset _kBadgeLabelBottomStart = Offset(20.25, 25.0);
+const Offset _kBadgeLabelBottomEnd = Offset(60.0, 25.0);
+
+/// Used to configure how the [NavigationDestination] positions its badge.
+enum BadgePosition {
+  /// Badge is positioned at the top start of the [NavigationDestination].
+  topStart,
+  /// Badge is positioned at the top end of the [NavigationDestination].
+  topEnd,
+  /// Badge is positioned at the bottom start of the [NavigationDestination].
+  bottomStart,
+  /// Badge is positioned at the bottom end of the [NavigationDestination].
+  bottomEnd,
+}
```

Comment:
wouldn't something like AlignmentGeometry make more sense? would support rtl out of the box and be more flexible

Reply:
This follows the material badge alignment naming scheme. 
Also start and end naming convention is implemented for RTL support rather left or right